enhancements/scorecard-storage.md

```diff
@@ -0,0 +1,258 @@
+ * scorecard cleanup would cleanup the test Pod as normal, thereby also removing the added storage sidecar container
+
+A successful scorecard execution of tests that use this storage feature
+would include a local directory of the collected test output, with test output being divided into subdirectories based on suite and test names
+
+### Risks and Mitigations
+
+Scorecard users that don't require persistent storage are not impacted by this proposed feature.
+
+The storage sidecar would use the wait-time value to timeout in
+the event that scorecard abnormally exited before it performed 
+its test pod cleanup.
+
+## Design Details
+
+### emptyDir Volumes
+Each scorecard test that specifies the storage feature will have a unique emptyDir Volume into which they can write test output. This ensures that each
+test can be write output in isolation from other tests.  These volumes
+will be removed as part of scorecard's test Pod cleanup.  The presence of
+a storage spec within the scorecard config file triggers the addtion of the storage sidecar.
+
+By using emptyDir volumes for this feature, the cluster doesn't have to have a storage class defined and volume cleanup is included as part of test Pod cleanup.
+
+### Errors
+
+### Test Output
+Scorecard test output is gathered from the emptyDir volumes using a sidecar container that continues to run after the test container completes. The extraction of test output is performed by exec/tar into the sidecar container.
+Scorecard test output is extracted from the 'sidecar' container using client-go APIs (e.g. exec).  
+
+Test output is organized by test name in sub-directories.  For example, gathered output by default is stored in the local directory where
+scorecard was executed, however the user can also specify a command line flag (e.g. --test-output) to have the
+test output stored at a custom location.  The gathered output appears as follows:
+```bash
+./test-output
+./test-output/custom-suite/example-test1/somefile
+./test-output/custom-suite/example-test2/anotherfile
+./test-output/custom-suite/example-test3/foo.log
+```
+This directory will be over-written if scorecard is executed multiple times, it is therefore up to the end user
+to perform data management of this test output directory as they require.
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy. Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
```

Comment:
I assume scorecard will have unit tests and the usual testing from the SDK repo right? And will QE also test this?

Reply:
thats correct, SDK would take ownership of this so the development of it would follow the standard SDK dev practices.